app/assets/javascripts/beak/widgets/ractives/code-editor.coffee

```diff
@@ -49,9 +50,48 @@ window.RactiveModelCodeComponent = Ractive.extend({
     $('#procedurenames-dropdown').trigger('chosen:updated')
     return
 
+  setupAutoComplete: (hintList) ->
+    CodeMirror.registerHelper('hintWords', 'netlogo', hintList)
+    editor = @findComponent('codeEditor').getEditor()
+    editor.on('keyup', (cm, event) =>
+      if not cm.state.completionActive and event.keyCode > 64 and event.keyCode < 91 and @get('autoCompleteStatus')
+        cm.showHint({completeSingle: false})
+    )
+    return
+
+  netLogoHintHelper: (cm, options) ->
+    cur = cm.getCursor()
+    token = cm.getTokenAt(cur)
+    to = CodeMirror.Pos(cur.line, token.end)
+    if token.string and /\S/.test(token.string[token.string.length - 1])
+      term = token.string
+      from = CodeMirror.Pos(cur.line, token.start)
+    else
+      term = ''
+      from = to
+    found = options.words.filter( (word) ->
+      if word.slice(0, term.length) is term
+        word
+    )
+    if found.length > 0
+      return { list: found, from: from, to: to }
+
+  autoCompleteWords: ->
+    autoCompleteWords = Object.keys(@getProcedureNames())
```

Comment:
I think it's asking for trouble/confusion to give an inner variable (`autoCompleteWords`) the same variable name as the function that it is declared within.  Please rename one of them.